pkg/netpol/eval/resources.go

```diff
@@ -582,29 +583,31 @@ func (pe *PolicyEngine) ConvertPeerNamedPort(namedPort string, peer Peer) (int32
 // the pe.podsMap in case of fake ingress-controller pods
 // or the pe.representativePeersMap in case of exposure-analysis peers
 func (pe *PolicyEngine) AddPodByNameAndNamespace(name, ns string, objLabels *k8s.SingleRuleLabels) (Peer, error) {
-	var nsLabels map[string]string
+	var nsLabels, podLabels map[string]string
 	if objLabels != nil {
 		nsLabels = objLabels.NsLabels
+		podLabels = objLabels.PodLabels
 	}
 
 	podStr := types.NamespacedName{Namespace: ns, Name: name}.String()
 	newPod := &k8s.Pod{
 		Name:      name,
 		Namespace: ns,
 		FakePod:   true,
+		Labels:    podLabels,
 	}
 	if err := pe.resolveSingleMissingNamespace(ns, nsLabels); err != nil {
 		return nil, err
 	}
-	if pe.exposureAnalysisFlag && newPod.Name == k8s.RepresentativePodName { // if exposure-analysis and this is not a fake ingress-controller
+	// if exposure-analysis and this is not a fake ingress-controller
+	if pe.exposureAnalysisFlag && strings.HasPrefix(newPod.Name, k8s.RepresentativePodName) {
```

Comment:
The expression `strings.HasPrefix(newPod.Name, k8s.RepresentativePodName) ` is repeated twice.. better have an explicit function `IsPodRepresentative()`

Reply:
done